src/Microsoft.AspNetCore.SignalR.Client/HubConnection.cs

```diff
@@ -45,29 +46,29 @@ public class HubConnection
             remove { _connection.Closed -= value; }
         }
 
+        public HubConnection(Uri url, IInvocationAdapter adapter)
```

Comment:
There will be one more - there should be a default `IInvocationAdapter` (presumably JSon?) - the user should not have to provide one. `IConnection` adds to the crud but this one is useful for testing.
On the other - once we have the one with the default invocation adapter - we can remove this one? In that case you would have to provide loggerfactory if you provide invocation adapter but it can be null